portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -116,25 +189,42 @@ private Query createCommentQuery(HttpServletRequest request) {
   }
 
   /**
-   * Converts a List into a JSON string using the Gson library.
+   * Translates a stream of comment texts to the specified language code and returns a
+   * stream of the translated comment texts.
    */
-  private String convertToJson(List list) {
+  private Stream<String> translateComments(Stream<String> comments, String languageCode) {
+    // Convert the stream of comment texts into a list since the translation API takes in lists.
+    List<String> commentTexts = comments.collect(Collectors.toList());
+
+    // Translate the list of comment texts.
+    Translate translate = TranslateOptions.getDefaultInstance().getService();
+    List<Translation> translations = translate.translate(commentTexts, 
+        Translate.TranslateOption.targetLanguage(languageCode), 
+        Translate.TranslateOption.format("text"));
+
+    // Map the translations to a stream of strings.
+    return translations.stream().map(Translation::getTranslatedText);
+  }
+
+  /**
+   * Converts a List into a JsonElement using the Gson library.
+   */
+  private JsonElement convertToJsonElement(List list) {
```

Comment:
If this method doesn't actually require any instance state, you can make this a static method.